packages/valory/skills/dynamic_nft_abci/models.py

```diff
@@ -32,6 +41,42 @@
 from packages.valory.skills.dynamic_nft_abci.rounds import DynamicNFTAbciApp
 
 
+DEFAULT_ADDRESS = "0x0000000000000000000000000000000000000000"
+DEFAULT_POINTS = 0
+VERIFICATION_POINTS = 100
+INSERT_ROW_INDEX = 2  # header is 1
+CACHE_EXPIRATION_SECONDS = 10
+CACHE_MAXSIZE = 128
+
+
+def timed_lru_cache(seconds: int, maxsize: Optional[int] = None):
+    """Timed cache"""
+
+    def wrapper_cache(func):
+        """Wrapper cache"""
+        func = lru_cache(maxsize=maxsize)(func)
+        func.lifetime = timedelta(seconds=seconds)
+        func.expiration = datetime.utcnow() + func.lifetime
+
+        @wraps(func)
+        def wrapped_func(*args, **kwargs):
+            """Wrapped func"""
+            if datetime.utcnow() >= func.expiration:
+                print("func.expiration lru_cache lifetime expired")
+                func.cache_clear()
+                func.expiration = datetime.utcnow() + func.lifetime
+
+            return func(*args, **kwargs)
+
+        # Add missing methods to wrapped function
+        wrapped_func.cache_clear = func.cache_clear
+        wrapped_func.cache_info = func.cache_info
+
+        return wrapped_func
+
+    return wrapper_cache
+
+
```

Comment:
This is a popular cache implementation we got from [here](https://realpython.com/lru-cache-python/).